include/nbl/builtin/hlsl/random/xoroshiro.hlsl

```diff
@@ -0,0 +1,77 @@
+
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+#define _NBL_BUILTIN_GLSL_RANDOM_XOROSHIRO_HLSL_INCLUDED_
+
+//#include <nbl/builtin/hlsl/math/functions.hlsl>
+#include <nbl/builtin/hlsl/cpp_compat/vector.hlsl>
+
+// TODO[Przemek]: include functions.hlsl instead
+uint32_t rotl(NBL_CONST_REF_ARG(uint32_t) x, NBL_CONST_REF_ARG(uint32_t) k)
+{
+   return (x<<k) | (x>>(32u-k));
+}
+
+namespace nbl
+{
+namespace hlsl
+{
+
+typedef uint2 xoroshiro64star_state_t;
+typedef uint2 xoroshiro64starstar_state_t;
+
+namespace impl
+{
+	uint2 xoroshiro64_state_advance(uint2 state)
+	{
+		state[1] ^= state[0];
+		state[0] = rotl(state[0], 26u) ^ state[1] ^ (state[1]<<9u); // a, b
+		state[1] = rotl(state[1], 13u); // c
+		
+		return state;
+	}
+}
```

Comment:
make it into a common case struct, remove the typedefs (store the state as a member in the base class)

Reply:
like that?
https://github.com/Devsh-Graphics-Programming/Nabla/pull/561/commits/72a5c39cdb64cecfeca4270a1c7ba0544f90af59

Reply:
you could have just inherited, but ok